core-web/libs/portlets/edit-ema/portlet/src/lib/components/dot-ema-dialog/dot-ema-dialog.component.ts

```diff
@@ -0,0 +1,192 @@
+        NgSwitch,
+        NgSwitchCase,
+        NgStyle,
+        SafeUrlPipe,
+        EmaFormSelectorComponent,
+        DialogModule,
+        DotSpinnerModule
+    ],
+    providers: [DotEmaDialogStore]
+})
+export class DotEmaDialogComponent {
+    @ViewChild('iframe') iframe: ElementRef<HTMLIFrameElement>;
+
+    @Output() customEvent = new EventEmitter<{ event: CustomEvent; payload: ActionPayload }>();
+    @Output() formSelected = new EventEmitter<{ identifier: string; payload: ActionPayload }>();
+
+    private readonly destroyRef$ = inject(DestroyRef);
+    private readonly store = inject(DotEmaDialogStore);
+
+    protected readonly dialogState = toSignal(this.store.dialogState$);
+
+    protected get ds() {
+        return this.dialogState();
+    }
+
+    /**
+     *
+     * @memberof DotEmaDialogComponent
+     */
+    resetDialog() {
+        this.store.resetDialog();
+    }
+
+    /**
+     * Add contentlet
+     *
+     * @param {ActionPayload} payload
+     * @memberof EditEmaEditorComponent
+     */
+    addContentlet(payload: ActionPayload): void {
+        this.store.openAddIframe({
+            containerId: payload.container.identifier,
+            acceptTypes: payload.container.acceptTypes ?? '*',
+            language_id: payload.language_id,
+            payload
+        });
+    }
+
+    /**
+     * Add Form
+     *
+     * @param {ActionPayload} _payload
+     * @memberof EditEmaEditorComponent
+     */
+    addForm(payload: ActionPayload): void {
+        this.store.openAddFormIframe(payload);
+    }
+
+    /**
+     * Add Widget
+     *
+     * @param {ActionPayload} payload
+     * @memberof EditEmaEditorComponent
+     */
+    addWidget(payload: ActionPayload): void {
+        this.store.openAddIframe({
+            containerId: payload.container.identifier,
+            acceptTypes: DotCMSBaseTypesContentTypes.WIDGET,
+            language_id: payload.language_id,
+            payload
+        });
+    }
+    /**
+     * Edit contentlet
+     *
+     * @param {ActionPayload} payload
+     * @memberof EditEmaEditorComponent
+     */
+    editContentlet(payload: Partial<ActionPayload>) {
+        this.store.openEditIframe({
+            inode: payload.contentlet.inode,
+            title: payload.contentlet.title
+        });
+    }
+
+    /**
+     * Create contentlet form
+     *
+     * @param {{ url: string; contentType: string }} { url, contentType }
+     * @memberof DotEmaDialogComponent
+     */
+    createContentlet({ url, contentType }: { url: string; contentType: string }) {
+        this.store.openCreateIframe({
+            url,
+            contentType
+        });
+    }
+
+    /**
+     * Create contentlet from palette
+     *
+     * @param {{
+     *         variable: string;
+     *         name: string;
+     *         payload: ActionPayload;
+     *     }} {
+     *         variable,
+     *         name,
+     *         payload
+     *     }
+     * @memberof DotEmaDialogComponent
+     */
```

Comment:
Why don't we try to type this and the params in the store, maybe we can have some kind of event type friendly API.